src/lib/expression/pqp_subquery_expression.cpp

```diff
@@ -43,11 +43,15 @@ std::string PQPSubqueryExpression::as_column_name() const {
 }
 
 bool PQPSubqueryExpression::_shallow_equals(const AbstractExpression& expression) const {
-  Fail("Can't compare PQPSubqueryExpressions");
+  const auto& other = static_cast<const PQPSubqueryExpression&>(expression);
+  return pqp == other.pqp && parameters == parameters;
 }
 
 size_t PQPSubqueryExpression::_on_hash() const {
-  Fail("PQPSubqueryExpressions can't, shouldn't and shouldn't need to be hashed");
```

Comment:
Talked to mrzzzrm about this who said that the possibility of this PR was not considered back when this statement was made.